### PR TITLE
Update gems

### PR DESCRIPTION
**Pull request by marcheing**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -34,7 +34,7 @@ gem 'sdoc', '~> 0.4.0', group: :doc
 gem 'rails-html-sanitizer', '~> 1.0'
 
 # Kalibro integration
-gem 'kalibro_client', '~> 1.2'
+gem 'kalibro_client'
```
Comment (danielkza) [inline Gemfile]: Isn't it a good idea to keep the kalibro_client version explicit? This way we don't have the risk of breaking configurations or processor while changing the client, and it makes it clear what version we tested with.

Comment (marcheing) [inline Gemfile]: I think the only reason would be to make it explicit what version we're using, because the gem won't be updated without running `gem update` or `bundle update`. Also, the installed version can be found in the `Gemfile.lock` file.

- by @pedroscocco

Comment (rafamanzo) [inline Gemfile]: Generally nobody is going to read the `Gemfile.lock` if he/she is not looking for a very nasty bug.

Going beyond, removing the restriction allows bundler to install lower versions as well, which for sure Prezento is not compatible.

Other positive feature from fixing the version just for bug fixes is that it makes who is going to update it to not let a update with new/modified features to get ignored among all other updates. By forcing to edit the `Gemfile`, the developer is expected to check what are the changes for this update and if they are compatible.

So in short, I strongly advise to keep the automatic updates just for minor updates.

Comment (diegoamc) [inline Gemfile]: I think this way is very clear about what version we used, the most recent.
I would keep the version only when this pattern does not apply. For example, we relasead a new version that broke something. So we need to force an older version to keep the deployed version of kalibro_processor working.

What do you think?

Comment (diegoamc) [inline Gemfile]: If we're going to follow what @rafamanzo and @danielkza suggested, I would do that for every gem on the `Gemfile`.
For example, the `pg` does not have a fixed version. It uses the most recent version.

Comment (rafamanzo) [inline Gemfile]: I want to stress this again: by removing the version this allows bundler to
resolve dependencies by installing for example kalibro_client 0.0.1 which
for sure is not compatible with Prezento.

That is a bug that we are introducing.

Again, it may be a bumer for us core developers this restriction when
working on updates. But think about independent collaborators. They have no
idea about what is happening with kalibro_client (and they are not expected
to).

Please, keep the version. If you are really decided to make it less
restrictive, keep it with `gem 'kalibro_client', '~> 1.3'`. This way it
will prohibit just major updates that break the API.

Em ter, 1 de set de 2015 às 10:43, Diego de Araújo Martinez Camarinha <
notifications@github.com> escreveu:

Comment (rafamanzo) [inline Gemfile]: Do we want to restrict bundler's capacity for solving conflicts by placing? I would say that in my opinion no.

We want to restrict kalibro_client because we know for sure that any version lower than 1.2 will break. I can't say that about pg which you mentioned.

Comment (diegoamc) [inline Gemfile]: So if a `pg` update, for example, has new/modified features, it's ok for us to ignore it?

Comment (danielkza) [inline Gemfile]: @diegoamc `pg` is a particular case in which we're pulling the gem manually, but our code is not the one that uses it; ActiveRecord is, and AFAIK, is the one that needs to look for certain features and use them as necessary. We don't actually depend on the `pg` API ourselves, so it doesn't really matter which version Bundler pulls. If at some point we have a feature that requires a particular minimum version then it would make sense to make that explicit.

And I like that using `~> 1.2` means 'any version higher than 1.2 but lower than 2.0'. It makes sure you cannot use a version missing the features we're assuming, but also allows newer versions as long as they are not supposed to have breaking changes (if we follow SemVer correctly).
